codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -299,16 +304,15 @@ protected void writeMiddlewareDocumentSerializerDelegator(
         if (payloadBinding.isPresent()) {
             MemberShape memberShape = payloadBinding.get().getMember();
             Shape payloadShape = model.expectShape(memberShape.getTarget());
-            ShapeType shapeType = payloadShape.getType();
             String memberName = symbolProvider.toMemberName(memberShape);
 
             Optional<MediaTypeTrait> mediaTypeTrait = payloadShape.getTrait(MediaTypeTrait.class);
             mediaTypeTrait.ifPresent(typeTrait -> writer.write("restEncoder.SetHeader(\"Content-Type\").String($S)",
                     typeTrait.getValue()));
 
-            if (shapeType == ShapeType.BLOB) {
+            if (payloadShape.isBlobShape()) {
                 writer.write("documentPayload = input.$L", memberName);
-            } else if (shapeType == ShapeType.STRING) {
+            } else if (payloadShape.isStringShape()) {
```

Comment:
Curious why `isBlobShape` method vs ShapeType.BLOB?  is one better to use than the other?

Reply:
Just syntactical sugar.